modules/unsupported/s3-geotiff/src/main/java/org/geotools/s3/S3Connector.java

```diff
@@ -116,24 +116,21 @@ public AmazonS3 getS3Client() {
 
             Properties prop = readProperties(s3Alias);
 
+            String endpoint = prop.getProperty(s3Alias + ".s3.endpoint");
+            AwsClientBuilder.EndpointConfiguration endpointConfiguration =
+                    new AwsClientBuilder.EndpointConfiguration(endpoint, region.getName());
+
             s3 =
                     AmazonS3ClientBuilder.standard()
                             .withCredentials(
                                     new AWSStaticCredentialsProvider(
                                             new BasicAWSCredentials(
                                                     prop.getProperty(s3Alias + ".s3.user"),
                                                     prop.getProperty(s3Alias + ".s3.password"))))
+                            .withEndpointConfiguration(endpointConfiguration)
+                            .withPathStyleAccessEnabled(true)
                             .build();
 
-            final S3ClientOptions clientOptions =
-                    S3ClientOptions.builder().setPathStyleAccess(true).build();
-            s3.setS3ClientOptions(clientOptions);
-            String endpoint = prop.getProperty(s3Alias + ".s3.endpoint");
-            if (!endpoint.endsWith("/")) {
-                endpoint = endpoint + "/";
```

Comment:
Not needed, as a leading slash is added to the resource path in `com.amazonaws.services.s3.internal.ServiceUtils.convertRequestToUrl(Request, boolean, boolean)`

Reply:
Indeed, and it looks like the code has also been simplified by proper use of the API.